source/pkger/pkger.go

```diff
@@ -0,0 +1,67 @@
+package pkger
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/golang-migrate/migrate/v4/source"
+	"github.com/golang-migrate/migrate/v4/source/httpfs"
+	"github.com/markbates/pkger/pkging"
+)
+
+func init() {
+	source.Register("pkger", &driver{})
+}
+
+// Instance is an implementation of http.FileSystem backed by an instance of
+// pkging.Pkger.
+type Instance struct {
+	pkging.Pkger
+
+	// Path is the relative path location of the migrations. It is passed to
+	// httpfs.PartialDriver.Init. If unset "/" is used as all paths are
+	// absolute.
+	Path string
+}
+
+// Open implements http.FileSystem.
+func (p *Instance) Open(name string) (http.File, error) {
```

Comment:
`Open()` should call `Init()`

Reply:
Does this still apply after reading the explanation of why there are two structs?